### PR TITLE
[Storage] Update generated extensible enums

### PR DESCRIPTION
**Pull request by ShivangiReja**
We're settling on a [shared pattern for extensible enums](https://gist.github.com/heaths/d105148428fe09a2631322b656f04ebb) and are [adding it to the Guidelines.](https://github.com/Azure/azure-sdk/pull/648/files)

This PR follows the official pattern in the generator.

### CODE REVIEW

**Review by tg-msft — APPROVED** [review]
Looks great!

File: sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts [hunk 1]
```diff
@@ -761,80 +774,61 @@ function generateEnumStrings(w: IndentWriter, model: IServiceModel, type: IEnumT
...
-            w.line(`public override bool Equals(object o) => o is ${enumFullName} other && Equals(other);`);
+            w.line(`public bool Equals(${enumFullName} other) => string.Equals(_value, other._value, System.StringComparison.Ordinal);`)
             w.line(``);
             w.line(`/// <summary>`);
-            w.line(`/// Get a hash code for the ${enumName}.`);
+            w.line(`/// Get a hash code for the <see cref="${enumName}"/>.`);
             w.line(`/// </summary>`);
             w.line(`/// <returns>Hash code for the ${enumName}.</returns>`);
-            w.line(`public override int GetHashCode() => _value.GetHashCode();`);
+            w.line(`[System.ComponentModel.EditorBrowsable((System.ComponentModel.EditorBrowsableState.Never))]`);
```
Comment (heaths) [inline sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts]: Spacing (at least in GH) seems off on this

File: sdk/storage/Azure.Storage.Queues/src/Generated/QueueRestClient.cs [hunk 2]
```diff
@@ -2681,93 +2681,84 @@ namespace Azure.Storage.Queues.Models
...
+        /// unavailable
         /// </summary>
-        /// <param name="other">The instance to compare to.</param>
-        /// <returns>True if they're equal, false otherwise.</returns>
-        public bool Equals(Azure.Storage.Queues.Models.GeoReplicationStatus other) => _value.Equals(other._value, System.StringComparison.InvariantCulture);
+        public static readonly Azure.Storage.Queues.Models.GeoReplicationStatus Unavailable = new GeoReplicationStatus(@"unavailable");
 
         /// <summary>
-        /// Check if two GeoReplicationStatus instances are equal.
+        /// Determines if two <see cref="GeoReplicationStatus"/> values are different.
```
Comment (heaths) [inline sdk/storage/Azure.Storage.Queues/src/Generated/QueueRestClient.cs]: Summary is reversed. Might be in the generator (just noticed this).

**Review by heaths — APPROVED** [review]

File: sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts (hunk 1 above)
Comment (ShivangiReja) [reply sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts]: Spacing seems good to me in VS (just removed extra `(` in `EditorBrowsable`)

**Review by heaths — APPROVED** [review]

**Pull request merged**